Copy "envFrom" from Velero server when creating maintenance jobs

evhan opened the pull request:
Thank you for contributing to Velero!

# Please add a summary of your change

When the Velero server creates a maintenance job, it copies its own `env` into the newly-created Pod spec. However, Velero's environment may be populated via `envFrom` in addition to `env`, in which case the maintenance job may fail since it will be missing potentially-important variables.

We ran into this while using the [OpenStack plugin](https://github.com/Lirt/velero-plugin-for-openstack), but the issue is not specific to that provider. We worked around the problem in the meantime by moving all variables that were previously stored in a Secret into the Deployment itself.

The documentation already says:

> Maintenance jobs will inherit the labels, annotations, toleration, nodeSelector, service account, image, environment variables, cloud-credentials etc. from Velero deployment.

This would appear to be covered by that, and the change is just bringing the implementation in line with the docs.

# Does your change fix a particular issue?

No existing issue in GitHub from what I can see.

# Please indicate you've done the following:

- [x] [Accepted the DCO](https://velero.io/docs/v1.5/code-standards/#dco-sign-off). Commits without the DCO will delay acceptance.
- [x] [Created a changelog file (`make new-changelog`)](https://velero.io/docs/main/code-standards/#adding-a-changelog) or comment `/kind changelog-not-required` on this PR.
- [ ] ~~Updated the corresponding documentation in `site/content/docs/main`.~~

Lyndon-Li commented:
This looks like a valid issue for both maintenance job and data mover pods, but by default Velero doesn't set `envFrom`, so we will not include it in 1.15 RC.2.

Lyndon-Li commented:
@evhan
Could you also help to fix the same problem for data mover pods? The code should be around functions `createBackupPod` and `createRestorePod`.

reasonerjt reviewed (changes requested):
@evhan Thanks for the PR.  We appreciate your contribution.

But could you also add a UT?

evhan commented:
> @evhan Thanks for the PR. We appreciate your contribution.
> 
> But could you also add a UT?

Sure. I'll also look at fixing the issue for the data mover pods. Would you like that in the same PR or a new one?

Lyndon-Li commented:
> Would you like that in the same PR or a new one?

All been included in the current PR will be fine. Thanks.

evhan commented:
> All been included in the current PR will be fine. Thanks.

No problem. I've added the same treatment for the data mover pods and added env/envFrom to the tests to make sure they're included. Hope that looks OK.

shubham-pampattiwar reviewed (commented):
Thank you @evhan for the contribution. CI is failing on `make update`. Can please you execute `make update` and update the pull request ?

evhan commented:
> Thank you @evhan for the contribution. CI is failing on `make update`. Can please you execute `make update` and update the pull request ?

Yeah, just noticed, sorry about that. Should be fixed now!

kaovilai reviewed (approved):
This LGTM. Thank you!

Lyndon-Li commented on pkg/exposer/generic_restore.go:
```diff
@@ -370,6 +370,7 @@ func (e *genericRestoreExposer) createRestorePod(ctx context.Context, ownerObjec
 					VolumeMounts:  volumeMounts,
 					VolumeDevices: volumeDevices,
 					Env:           podInfo.env,
+					EnvFrom:       podInfo.envFrom,
```
Please do the same for `createBackupPod` (exposer/csi_snapshot.go)

evhan replied:
Done. :man_facepalming:
